src/main/cli.py

```diff
@@ -6,280 +6,224 @@
-    if len(words) < 2:
-        print(f"{RED}\
-ERROR: one-word message is not informative, add more details to {segment}!\n\
-{DEFAULT}")
-        IS_EXIT_1 = True
+    if len(words) < MIN_WORDS:
+        print(
+            f"{RED}ERROR: one-word message is not informative, add more details to {section}!{OFF}\n")
+        is_valid = False
+    return is_valid
 
 
-def check_prefix(msg: str, segment=""):
+def is_valid_prefix(msg: str, section="") -> bool:
     """
-    Check if a prefix of the message is valid.
+    Validate the prefix of the message.
 
-    If tprefix of the message is invalid
-    display appropriate error and hint and continue with other checks.
+    If validation failed, display appropriate error and hint and continue with other checks.
 
     Args:
         msg (str): The part of commit mesage(subject line or body).
-        segment (str, optional): Segment to show in error message.
-        Defaults to empty string.
+        section (str, optional): The section where the error occurred. Defaults to empty string.
+    Returns:
+        bool: The result of the validation.
     """
-    global IS_EXIT_1
-    is_valid_prefix = msg.lstrip().startswith(tuple(PREFIXES))
+    global default_prefixes
+    is_valid = True
+    is_valid_prefix = msg.lstrip().startswith(tuple(default_prefixes))
     if msg[0].islower():
-        print(f"{RED}ERROR: capitalise {segment}!\n{DEFAULT}")
-        IS_EXIT_1 = True
+        print(f"{RED}ERROR: capitalise {section}!{OFF}\n")
+        is_valid = False
     if not is_valid_prefix:
         nl = "...\n       "
         print(f"{RED}\
-ERROR: wrong {segment} prefix!\n{YELLOW}\
-HINT:  you can add new prefixes as an {BLUE}args: {YELLOW}in {BLUE}\
-.pre-commit-config.yaml\n{YELLOW}\
+ERROR: wrong {section} prefix!\n{YELLOW}\
+HINT:  you can add new prefixes as an {BLUE}args: {YELLOW}in {BLUE}.pre-commit-config.yaml\n{YELLOW}\
        otherwise, replace prefix with one of the following options:\n{YELLOW}\
-       {nl.join(PREFIXES)}...{DEFAULT}")
-        IS_EXIT_1 = True
+       {nl.join(default_prefixes)}...{OFF}")
+        is_valid = False
+    return is_valid
 
 
-def check_ending(msg: str, segment=""):
+def is_valid_ending(msg: str, section="") -> bool:
     """
-    Check whether the message specific part ends with a dot or not.
+    Check whether the message ends with a dot or not.
 
-    If the message ends with a dot or not
-    display appropriate error and hint and continue with other checks.
+    If the message ends with a dot display appropriate error and hint and continue validation.
 
     Args:
         msg (str): The part of commit mesage(subject line or body).
-        segment (str, optional): Segment to show in error message.
-        Defaults to empty string.
+        section (str, optional): The section where the error occurred. Defaults to empty string.
     """
-    global IS_EXIT_1
+    is_valid = True
     if msg.rstrip().endswith("."):
```

Comment:
Same here

Reply:
If the msg is empty the logic of the hook is not starts